tests/flows/native_source_native_destination.go

```diff
@@ -0,0 +1,216 @@
+	valueToReturn           = big.NewInt(0).Div(valueToReceive, big.NewInt(4))
+	multiplyOnReceive       = true
+)
+
+/**
+ * Deploy a native token source on the primary network
+ * Deploys a native token destination to Subnet A
+ * Bridges C-Chain native tokens to Subnet A
+ * Bridge back tokens from Subnet A to C-Chain
+ */
+func NativeSourceNativeDestination(network interfaces.Network) {
+	cChainInfo := network.GetPrimaryNetworkInfo()
+	subnetAInfo, _ := teleporterUtils.GetTwoSubnets(network)
+	fundedAddress, fundedKey := network.GetFundedAccountInfo()
+
+	ctx := context.Background()
+
+	// Deploy an example WAVAX on the primary network
+	wavaxAddressA, wavaxA := utils.DeployExampleWAVAX(
+		ctx,
+		fundedKey,
+		cChainInfo,
+	)
+
+	// Deploy an example WAVAX on the primary network
+	wavaxAddressB, _ := utils.DeployExampleWAVAX(
+		ctx,
+		fundedKey,
+		subnetAInfo,
+	)
+
+	// Create a NativeTokenSource on the primary network
+	nativeTokenSourceAddress, nativeTokenSource := utils.DeployNativeTokenSource(
+		ctx,
+		fundedKey,
+		cChainInfo,
+		fundedAddress,
+		wavaxAddressA,
+	)
+
+	// Deploy an NativeTokenDestination to Subnet A
+	nativeTokenDestinationAddress, nativeTokenDestination := utils.DeployNativeTokenDestination(
+		ctx,
+		subnetAInfo,
+		fundedAddress,
+		cChainInfo.BlockchainID,
+		nativeTokenSourceAddress,
+		wavaxAddressB,
+		initialReserveImbalance,
+		decimalsShift,
+		multiplyOnReceive,
+	)
+
+	// Generate new recipient to receive bridged tokens
+	recipientKey, err := crypto.GenerateKey()
+	Expect(err).Should(BeNil())
+	recipientAddress := crypto.PubkeyToAddress(recipientKey.PublicKey)
+
+	// Send tokens from C-Chain to recipient on subnet A that don't fully collateralize bridge
+	{
+		input := nativetokensource.SendTokensInput{
+			DestinationBlockchainID:  subnetAInfo.BlockchainID,
+			DestinationBridgeAddress: nativeTokenDestinationAddress,
+			Recipient:                recipientAddress,
+			PrimaryFee:               big.NewInt(0),
+			SecondaryFee:             big.NewInt(0),
+			RequiredGasLimit:         utils.DefaultNativeTokenRequiredGasLimit,
+		}
+
+		receipt, bridgedAmount := utils.SendNativeTokenSource(
+			ctx,
+			cChainInfo,
+			nativeTokenSource,
+			input,
+			valueToSend,
+			fundedKey,
+		)
+		scaledBridgedAmount := teleporterUtils.BigIntMul(bridgedAmount, tokenMultipler)
+
+		receipt = network.RelayMessage(
+			ctx,
+			receipt,
+			cChainInfo,
+			subnetAInfo,
+			true,
+		)
+
+		utils.CheckNativeTokenDestinationMint(
+			ctx,
+			subnetAInfo,
+			nativeTokenDestination,
+			recipientAddress,
+			receipt,
+			big.NewInt(0),
+			big.NewInt(0),
+		)
+		utils.CheckNativeTokenDestinationCollateralize(
+			ctx,
+			nativeTokenDestination,
+			receipt,
+			scaledBridgedAmount,
+			teleporterUtils.BigIntSub(initialReserveImbalance, scaledBridgedAmount),
+		)
+	}
+
+	// Send tokens from C-Chain to recipient on subnet A that over-collateralize bridge
```

Comment:
Super nit: Technically the contracts are not "over-collateralized", I think you mean that the amount exceeds the remaining reserve imbalance